python/console/console_sci.py

```diff
@@ -437,7 +438,7 @@ def runFile(self, filename):
 
         try:
             # Run the file
-            self.runCommand("exec(Path('{0}').read_text())".format(filename), skipHistory=True)
+            self.runCommand("exec(compile(Path('{0}').read_text(), '{1}', 'exec'))".format(filename, override_file_name or filename), skipHistory=True)
```

Comment:
wouldn't there be issues if the filename has a single-quote character ?

Reply:
@rouault 

True (although not introduced in this change).

Do you know any reliable inbuilt way to escape this string for use in eval? If we're doing this right then a simple `.replace( "'", "\\'")` won't cut it...

Reply:
Looks like elsewhere we use QgsProcessingUtils::stringToPythonLiteral, even outside of processing. I'll do that here.

Reply:
Fixed